tlv/record.go

```diff
@@ -17,6 +17,10 @@ type Type uint64
 // slice containing the encoded data.
 type TypeMap map[Type][]byte
 
+// Blob is a type alias for a byte slice. It's used to indicate that a slice of
+// bytes is actually an encoded TLV stream.
+type Blob = []byte
```

Comment:
If there are different semantics here that are being assumed, I'd recommend not making it a simple alias and actually make a newtype (dropping the =)

Reply:
In this case using the type alias is useful, as then I can put it in a `tlv.RecordT` as a primitive type. Otherwise, you'd need the record method wrapper.

Reply:
We should just make RecordT a primitive type, no? I just want to stress that type aliases are not useful for anything other than giving the reader a hint on how something is used, it assures no correct api usage the way that a newtype wrapper does.